client/src/components/Slider/Slide.jsx

```diff
@@ -0,0 +1,42 @@
+import React, { useRef, useState, useEffect } from 'react';
+import { easeInOutSine } from '../../easings';
+
+const Slide = ({ src, alt, position = 0 }) => {
+  const [left, setLeft] = useState(0); // position of slide on x axis
+  const ref = useRef(null); //
+  const firstUpdate = useRef(true);
+
+  useEffect(() => {
+    // handling the componentDidMount
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      setLeft((left) => ref.current.offsetWidth * position);
+      return;
+    }
+
+    // handling the componentDidUpdate
```

Comment:
ogólnie komentarze dodawaj tylko naprawdę jak musisz.
Znacznie lepiej jest wydzielić  kod do oddzielnej funkcji , której nazwa de facto opisuje to co robisz i zastępuje komentarz